### PR TITLE
Ju 312 java 21 migration

### PR DESCRIPTION
**Pull request by jaddison-ch**
### Work checklist

* [x] Bug fix
* [x] New feature
* [x] Breaking change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/email/EmailSendMessageProducer.java [hunk 1]
```diff
@@ -15,10 +16,13 @@
...
 
 	private static final String EXCEPTION_MESSAGE = "Kafka 'email-send' message could not be sent for appeal with penalty reference - %s";
-	
-    private final EmailSendMessageFactory emailSendAvroSerializer;
-    private final EmailSendKafkaProducer emailSendKafkaProducer;
+	@Autowired
+    EmailSendMessageFactory emailSendAvroSerializer;
 
+    @Autowired
+    EmailSendKafkaProducer emailSendKafkaProducer;
```
Comment (hwatkins-ch) [inline src/main/java/uk/gov/companieshouse/email/EmailSendMessageProducer.java:23]: Private?

File: src/main/java/uk/gov/companieshouse/service/AppealService.java [hunk 2]
```diff
@@ -27,24 +27,24 @@
...
-    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APP_NAMESPACE);
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APPLICATION_NAME_SPACE);
     private static final String USER_ID = "user_id";
     private static final String APPEAL_ID = "appeal_id";
     private static final String PENALTY_REF = "penalty_reference";
     private static final String COMPANY_NUMBER = "company_number";
 
     @Autowired
-    private AppealMapper appealMapper;
+     AppealMapper appealMapper;
```
Comment (hwatkins-ch) [inline src/main/java/uk/gov/companieshouse/service/AppealService.java:37]: All these can be private?

File: src/main/java/uk/gov/companieshouse/service/CompanyProfileService.java [hunk 3]
```diff
@@ -6,29 +6,29 @@
...
 public class CompanyProfileService {
 	
-    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APP_NAMESPACE);	
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APPLICATION_NAME_SPACE);	
 
     private static final UriTemplate GET_COMPANY_URI = new UriTemplate("/company/{companyNumber}");
 
     @Autowired
-    private ApiClientService apiClientService;
+    InternalApiClient internalApiClient;
```
Comment (hwatkins-ch) [inline src/main/java/uk/gov/companieshouse/service/CompanyProfileService.java:25]: Private?

File: src/main/java/uk/gov/companieshouse/service/EmailService.java [hunk 4]
```diff
@@ -28,13 +31,15 @@
...
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APP_NAMESPACE);
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APPLICATION_NAME_SPACE);
 
     // This email address is supplied only to satisfy Avro contract.
     private static final String TOKEN_EMAIL_ADDRESS = "lfp-appeals@ch.gov.uk";
 
-    private final EmailSendMessageProducer producer;
-    private final CompanyProfileService companyProfileService;
+    @Autowired
```
Comment (hwatkins-ch) [inline src/main/java/uk/gov/companieshouse/service/EmailService.java:39]: All these can be private

File: src/main/java/uk/gov/companieshouse/controller/AppealController.java [hunk 5]
```diff
@@ -37,8 +37,10 @@
 @RequestMapping("/companies")
 public class AppealController {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APP_NAMESPACE);
-    private final AppealService appealService;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppealApplication.APPLICATION_NAME_SPACE);
+
+    @Autowired
+    AppealService appealService;
```
[the remark this answers is not in the document]
Comment (jaddison-ch) [reply src/main/java/uk/gov/companieshouse/controller/AppealController.java:43]: this has been amended as requested

File: src/main/java/uk/gov/companieshouse/email/EmailSendMessageProducer.java [hunk 6]
```diff
@@ -15,10 +16,13 @@
 public class EmailSendMessageProducer {
 
 	private static final String EXCEPTION_MESSAGE = "Kafka 'email-send' message could not be sent for appeal with penalty reference - %s";
-	
-    private final EmailSendMessageFactory emailSendAvroSerializer;
-    private final EmailSendKafkaProducer emailSendKafkaProducer;
+	@Autowired
+    EmailSendMessageFactory emailSendAvroSerializer;
```
[the remark this answers is not in the document]
Comment (jaddison-ch) [reply src/main/java/uk/gov/companieshouse/email/EmailSendMessageProducer.java:20]: this has been amended as requested

File: src/main/java/uk/gov/companieshouse/email/EmailSendMessageProducer.java (hunk 1 above)
Comment (jaddison-ch) [reply src/main/java/uk/gov/companieshouse/email/EmailSendMessageProducer.java:23]: this has been amended as requested

File: src/main/java/uk/gov/companieshouse/service/AppealService.java (hunk 2 above)
Comment (jaddison-ch) [reply src/main/java/uk/gov/companieshouse/service/AppealService.java:37]: this has been amended as requested

File: src/main/java/uk/gov/companieshouse/service/EmailService.java (hunk 4 above)
Comment (jaddison-ch) [reply src/main/java/uk/gov/companieshouse/service/EmailService.java:39]: this has been amended as requested

**Pull request merged**